policies/wip/intoto-envelope.dog

```diff
@@ -0,0 +1,18 @@
+
+// Use the built in pattern to check the fields look right
+pattern format-check = intoto::envelope
+
+// Prepare parameter for intoto::verify-envelope
+pattern attesters = [
+    {
+        name: "RHTAP Chains Signing Key", // Does the name matter?
+        public_key: "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEZP/0htjhVt2y0ohjgtIIgICOtQtAnaYJRuLprwIv6FDhZ5yFjYUEtsmoNcW7rx2KM6FOXGsCX3BNc7qhHELT+g=="
+    }
+]
+
+// The sha of the manifest is the container image sha, so in theory this could work.
+// FIXME: Actually it throws errors about being unable to parse the blob.
+pattern blob = *data::from<"manifest.json">
+pattern sig-check = intoto::verify-envelope<attesters, blob>
```

Comment:
No, this can't be right either.

There are three parameters at play: attesters, in-toto envelope, and the software artifact. `blob` I think is the software artifact which in this case is the image manifest(?). The in-toto envelope is the input.

I get the same error you mentioned. Out of ideas.

Reply:
I can't get `cosign verify-blob` to work here, but `verify-blob-attestation` works wonderfully:

```
🐚 cosign verify-blob-attestation --key cosign.pub --signature input/attestation.json data/manifest.json  --insecure-ignore-tlog --type slsaprovenance
WARNING: Skipping tlog verification is an insecure practice that lacks of transparency and auditability verification for the blob attestation.
Verified OK
```